src/main/kotlin/me/zeroeightsix/kami/feature/module/DiscordStatus.kt

```diff
@@ -0,0 +1,119 @@
+            "Rockin' version " + KamiMod.MODVER,
+            getOperatingSystemMessage(),
+            "YES!!!!!!",
+            "086 is NOT A PRIME NUMBER!",
+            "3 is a PRIME NUMBER",
+            "Also try our sister mod KAMI BLUE!"
+    )
+
+    override fun onEnable() {
+        rpcEnabled = true
+        initDiscord()
+    }
+
+    override fun onDisable() {
+        DiscordRPC.discordShutdown()
+        rpcEnabled = false
+        println("downdoot")
+    }
+
+    @EventHandler
+    private val updateListener =
+            Listener<TickEvent.Client.InGame>({
+                if (lastUpdate == 0L) lastUpdate = System.currentTimeMillis()
+                if ((lastUpdate + 15000L <= System.currentTimeMillis()) && rpcEnabled) { // 15 seconds, should be good.
+                    DiscordRPC.discordRunCallbacks()
+                    var bottomString = ""
+                    var topString = ""
+                    if (message && server) bottomString = ("127.0.0.1 (ez doxx)") + " | " + messages.random()
+                    else if (message) bottomString = messages.random()
+                    else if (server) bottomString = ("127.0.0.1 (ez doxx)")
+
+                    if (version && username) topString = (KamiMod.MODVER) + " | Notch"
+                    else if (version) topString = messages.random()
+                    else if (username) topString = ("Notch")
+
+
+                    val presence = DiscordRichPresence.Builder(bottomString) // This is the bottom half of the RPC with server ip and funny message
+                    presence.setDetails(topString) // This is the top half with modver and username
+                    presence.setBigImage("kami", "kamiclient.com") // Image key (kami) and text when you scroll over it (kamiclient.com)
+                    DiscordRPC.discordUpdatePresence(presence.build())
+                    lastUpdate += 15000L
+                }
+
+
+            })
+
+    // Yes, its the same code but again. Yes, I know its bad practice but at the moment I don't care.
+    fun initDiscord() {
+        var bottomString = ""
+        var topString = ""
+        if (message && server)  bottomString = ("127.0.0.1 (ez doxx)") + " | " + messages.random()
+        else if (message) bottomString = messages.random()
+        else if (server) bottomString = ("127.0.0.1 (ez doxx)")
+
+        if (version && username) topString =  (KamiMod.MODVER) + " | Notch"
+        else if (version) topString = messages.random()
+        else if (username) topString = ("Notch")
+
+        val handlers = DiscordEventHandlers.Builder().setReadyEventHandler { user: DiscordUser ->
+            val presence = DiscordRichPresence.Builder(bottomString)
+            presence.setDetails(topString)
+            DiscordRPC.discordUpdatePresence(presence.build())
+        }.build()
+
+        DiscordRPC.discordInitialize("753664640789118999", handlers, false);
+        DiscordRPC.discordRegister("753664640789118999", "");
```

Comment:
What's this magic number? I'm assuming it's some kind of ID for the rich presence 'application', or whatever discord calls them - can you create a `private const val` for it?

Reply:
Should be user customizable imo

Reply:
Yes, it's the application id. I'll create a variable for it.

Reply:
If you do that, you need to make the names for the image icons customizable as well, since those correlate with the uploaded names on your discord application (not the discord client, the one you create for your rpc that gives you the ID)